lib/crawly/worker.ex

```diff
@@ -76,16 +76,26 @@ defmodule Crawly.Worker do
              spider_name: atom(),
              response: HTTPoison.Response.t(),
              result: {:ok, {response, spider_name}} | {:error, term()}
-  defp get_response({request, spider_name}) do
+  def get_response({request, spider_name}) do
     # check if spider-level fetcher is set. Overrides the globally configured fetcher.
     # if not set, log warning for explicit config preferred,
     # get the globally-configured fetcher. Defaults to HTTPoisonFetcher
+
     {fetcher, options} =
-      Crawly.Utils.get_settings(
-        :fetcher,
-        spider_name,
-        {Crawly.Fetchers.HTTPoisonFetcher, []}
-      )
+      case Crawly.Utils.get_settings(
+             :fetcher,
+             spider_name,
+             {Crawly.Fetchers.HTTPoisonFetcher, []}
+           ) do
+        {module, args} ->
+          {module, args}
+
+        {module} ->
+          {module, nil}
+
+        module ->
```

Comment:
also, add in an `is_atom` guard for the module.

as a side note, this unwrapping logic can probably be moved to a utils function, as it is used by other workers as well.

Reply:
also, can you add a default case which raises an invalid format error

Reply:
Done